zkml/src/tensor.rs

```diff
@@ -1490,6 +1468,61 @@ where
             og_shape: self.og_shape.clone(),
         })
     }
+    /// Makes a [`Tensor`] that is a batch of lower triangular matrices.
+    /// - `matrix_dim` the number specifying the dimensions of each individual matrix (lower triangular amtrix must be square)
+    /// - `num_matrices` specifies how many matrices to make
+    /// - `diag` specifies the "offset" for the diagonal, an offset of `1` means we keep two `1`s on the first row instead of 1, and offset of `-1` means all `zeroes`
```

Comment:
is this the same behavior as pytorch ? if so can you just link it ? Or just put their description, which is, at least in my head, clearer
```
The argument [diagonal](https://docs.pytorch.org/docs/stable/generated/torch.diagonal.html#torch.diagonal) controls which diagonal to consider. If [diagonal](https://docs.pytorch.org/docs/stable/generated/torch.diagonal.html#torch.diagonal) = 0, all elements on and below the main diagonal are retained. A positive value includes just as many diagonals above the main diagonal, and similarly a negative value excludes just as many diagonals below the main diagonal. 
```

If **NOT**, then we should change the name because it's confusing and can you give an example ?